src/hooks/useRecommendationsObserver.ts

```diff
@@ -0,0 +1,57 @@
+import { useEffect } from 'react';
+import { Nullable } from '@constructor-io/constructorio-client-javascript/lib/types';
+import ConstructorIO from '@constructor-io/constructorio-client-javascript';
+import { Section } from '../types';
+
+function useRecommendationsObserver(
+  menuIsOpen: boolean,
+  sections: Section[],
+  constructorIO: Nullable<ConstructorIO>,
+  callback: (
+    target: HTMLElement,
+    sections: Section[],
+    constructorIO: Nullable<ConstructorIO>
+  ) => void
+) {
+  // Get refs for each section
+  const refs = sections
+    .filter((section) => section.type === 'recommendations')
+    .map((section) => section.ref);
+
+  useEffect(() => {
+    const observer = new IntersectionObserver(
+      (entries) => {
+        // For each section, check if it's intersecting
+        entries.forEach((entry) => {
+          if (entry.isIntersecting) {
+            // Call the callback, which will be trackRecommendationView in our case
+            callback(entry.target as HTMLElement, sections, constructorIO);
```

Comment:
are we only passing `trackRecommendationView` as our `callback`? 
 - if so, I'd prefer that we remove this comment and update the argument name to `trackRecommendationView` to make this explicit, then remove this comment.
 - else, let's remove this comment or update it to explain other use cases

Reply:
https://github.com/Constructor-io/constructorio-ui-autocomplete/pull/107/commits/55c168e54f3bc1568827cba615592e193f2fc3c1#diff-54d8f2b4755e1b9a3357013b05b24a47635a9fbae6dc1ab0116aa0dcaf69aa05R44